README.md

```diff
@@ -3,18 +3,281 @@
+|    list\<bool>     |      []bool      |     list      |     List\<Boolean>     |
+|     list\<i32>     |     []int32      |     list      |     List\<Integer>     |
+|     list\<i64>     |     []int64      |     list      |      List\<Long>       |
+|   list\<double>    |    []float64     |     list      |     List\<Double>      |
+|   list\<string>    |     []string     |     list      |     List\<String>      |
+|  map\<bool, bool>  |  map[bool]bool   |      map      | Map\<Boolean, Boolean> |
+|  map\<bool, i32>   |  map[bool]int32  |      map      | Map\<Boolean, Integer> |
+|  map\<bool, i64>   |  map[bool]int64  |      map      |  Map\<Boolean, Long>   |
+| map\<bool, double> | map[bool]float64 |      map      | Map\<Boolean, Double>  |
+| map\<bool, string> | map[bool]string  |      map      | Map\<Boolean, String>  |
 
-## reference
+**Important notes**:
+1. The list of map types is not exhaustive and includes only tested cases.
+Please do not use keys and values with **i8**, **i16** and **binary**.
+2. Currently only the thrift type and java type mappings documented in the table are supported.
+More mappings(e.g. **bool** <-> **boolean**) would be supported in the future.
+Please see [issue](https://github.com/kitex-contrib/codec-dubbo/issues/46).
+
+### Kitex-Dubbo Interoperability
+
+1. **kitex -> dubbo**  
+Writing **api.thrift** based on existing **dubbo Interface API** and [**Type Mapping Table**](#type-mapping). Then using
+latest kitex command tool and thriftgo to generate stub code.
+2. **dubbo -> kitex**  
+Writing dubbo client code based on existing **api.thrift** and [**Type Mapping Table**](#type-mapping).
+
+## Getting Started
+
+[**Concrete sample**](https://github.com/kitex-contrib/codec-dubbo/tree/main/samples//helloworld/).
+
+### Prerequisites
+
+```shell
+# compile kitex cmd tool
+mkdir ~/dubbo && cd ~/dubbo
+git clone https://github.com/Lvnszn/kitex.git
+cd kitex
+git checkout feature_idl_for_hessian
+go get github.com/cloudwego/thriftgo@main
+go mod tidy
+cd tool/cmd/kitex
+go build
+
+# compile thriftgo
+cd ~/dubbo
+git clone https://github.com/cloudwego/thriftgo.git
+cd thriftgo
+go build
+```
+
+### Generating kitex stub codes
+
+```shell
+export PATH=~/dubbo/thriftgo:$PATH
+mkdir ~/kitex-dubbo-demo && cd ~/kitex-dubbo-demo
+go mod init kitex-dubbo-demo
+cat > api.thrift << EOF
+namespace go hello
+
+struct GreetRequest {
+    1: required string req,
+}(JavaClassName="org.cloudwego.kitex.samples.api.GreetRequest")
+
+struct GreetResponse {
+    1: required string resp,
+}(JavaClassName="org.cloudwego.kitex.samples.api.GreetResponse")
+
+service GreetService {
+    string Greet(1: string req)
+    GreetResponse GreetWithStruct(1: GreetRequest req)
+}
+
+EOF
+~/dubbo/kitex/tool/cmd/kitex/kitex -module kitex-dubbo-demo -thrift template=slim -service GreetService -protocol Hessian2 ./api.thrift
+```
+
+Important Notes:
+1. JavaClassName of struct in **api.thrift** must be consistent with the target one.
```

Comment:
Not necessary for now?

Reply:
JavaClassName of struct must match the target one otherwise **hessian2.Decoder** would return a ```map[string]interface{}``` representing the target one.